src/compiler/mir/transform/function.rs

```diff
@@ -38,7 +41,11 @@ impl<'a> FuncTransformer<'a> {
 
         // Add the parameters of the function to the set of variables
         func.params.iter().for_each(|p| {
-            self.mir.arg(p.name, p.context().ty(), p.context().span());
+            let ty = self
+                .project
+                .add_type(p.context().ty())
```

Comment:
Is it to separate the construction of the type table from the transformation into MIR?  That is, have a separate traversal step that adds all types to the type table before the MIR transformation?